tests/unit/select_test.coffee

```diff
@@ -0,0 +1,131 @@
+select = null
+dispatcher = null
+
+module "[Unit] Select unit tests",
+  setup: ->
+    Ember.run ->
+      dispatcher = Ember.EventDispatcher.create()
+      dispatcher.setup()
+      select = Ember.Widgets.SelectComponent.create
+        content: ['foo', 'bar', 'barca', 'baz']
+
+  teardown: ->
+    Ember.run ->
+      dispatcher.destroy()
+      select.destroy()
+
+test 'Test continuous queries in a row', ->
+  expect 5
+
+  select.set 'query', 'ba'
+  equal(select.get('filteredContent')[0], 'bar')
+  equal(select.get('filteredContent')[1], 'barca')
+  equal(select.get('filteredContent')[2], 'baz')
+
+  select.set 'query', 'bar'
+  equal(select.get('filteredContent')[0],'bar')
+  equal(select.get('filteredContent')[1],'barca')
+
+test 'Test using array proxy', ->
+  expect 2
+
+  data = Ember.ArrayProxy.create({
+    content: ['red', 'reddit', 'green', 'blue']
+  });
+  select.set 'content', data
+
+  select.set 'query', 're'
+  equal(select.get('filteredContent')[0], 'red')
+  equal(select.get('filteredContent')[1], 'reddit')
+
+test 'Test sorted filter', ->
+  expect 3
+
+  data = ['reddit', 'red', 'green', 'blue']
+  select.set 'content', data
+
+  select.set 'query', 'r'
+  equal(select.get('sortedFilteredContent')[0], 'green')
+  equal(select.get('sortedFilteredContent')[1], 'red')
+  equal(select.get('sortedFilteredContent')[2], 'reddit')
+
+test 'Test keyboard interaction', ->
+  expect 10
+  selectedText = null
+
+  append(select)
+
+  selectComponent = select.$()
+
+  validateDropdownVisible = (messageVisible) ->
+    ok isVisible(find '.ember-select-results', selectComponent),
+      messageVisible
+
+  validateDropdownHidden = (messageHidden) ->
+    ok isHidden(find '.ember-select-results', selectComponent),
+      messageHidden
+
+  validateFocus = (messageFocus) ->
+    ok isFocused(selectComponent,selectComponent), messageFocus
+
+  validateDropdownHidden('Dropdown list should not exist at the beginning')
+  selectComponent.focus()
+  wait().then ->
```

Comment:
Actually, `keyEvent` is an asynchronous helper, which means it can be chained without using `then`, `wait` or `andThen`

So we would probably better writing

```cs
append()
...
andThen() ->
    validateDropdownHidden
    selectComponent.focus()

pressEnter selectComponent
andThen ->
    validateDropdownVisible

pressDown selectComponent
andThen ->
    ....
```